label_studio/core/migration_helpers.py

```diff
@@ -0,0 +1,101 @@
+import logging
+from typing import Callable, Tuple
+
+from core.redis import start_job_async_or_sync
+from django.conf import settings
+from django.db import connection
+
+logger = logging.getLogger(__name__)
+
+
+def execute_sql_job(*, migration_name: str, sql: str, apply_on_sqlite: bool = False, reverse: bool = False) -> None:
+    from core.models import AsyncMigrationStatus
+
+    if not reverse:
+        migration, created = AsyncMigrationStatus.objects.get_or_create(
+            name=migration_name,
+            defaults={'status': AsyncMigrationStatus.STATUS_STARTED},
+        )
+        if not created and migration.status == AsyncMigrationStatus.STATUS_FINISHED:
+            logger.info(f'Migration {migration_name} already executed with status FINISHED')
+            return
+        if migration.status == AsyncMigrationStatus.STATUS_SCHEDULED:
+            migration.status = AsyncMigrationStatus.STATUS_STARTED
+            migration.save()
+
+        try:
+            if connection.vendor == 'sqlite' and not apply_on_sqlite:
+                logger.info('SQLite detected; skipping SQL execution as requested')
+            else:
+                with connection.cursor() as cursor:
+                    cursor.execute(sql)
+            migration.status = AsyncMigrationStatus.STATUS_FINISHED
+            migration.save()
+        except Exception as e:
+            logger.error(f'Migration {migration_name} failed: {e}')
+            migration.status = AsyncMigrationStatus.STATUS_ERROR
+            if not migration.meta:
+                migration.meta = {}
+            migration.meta['error'] = str(e)
+            migration.save()
+            raise
+    else:
+        # Reverse path: don't create/update AsyncMigrationStatus. Just run SQL.
```

Comment:
Since this functionality is part of the most basic architecture, I would cover these two functions, execute_sql_job and make_sql_migration, more thoroughly:

https://app.codecov.io/gh/HumanSignal/label-studio/pull/8767?src=pr&el=tree&filepath=label_studio%2Fusers%2Fadmin.py&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=HumanSignal#diff-bGFiZWxfc3R1ZGlvL3VzZXJzL2FkbWluLnB5